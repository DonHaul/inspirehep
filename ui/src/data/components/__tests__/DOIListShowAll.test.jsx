import React from 'react';
import { render } from '@testing-library/react';
import { fromJS } from 'immutable';
import DOIListShowAll from '../DOIListShowAll';

describe('DOIListShowAll', () => {
  it('renders correctly with default props', () => {
    const { asFragment } = render(<DOIListShowAll />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with data and other dois', () => {
    const dois = fromJS([
      { material: 'data', value: 'doi1' },
      { material: 'data', value: 'doi2' },
      { material: 'article', value: 'doi3' },
    ]);
    const { asFragment } = render(<DOIListShowAll dois={dois} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with dataDois only', () => {
    const dois = fromJS([
      { material: 'data', value: 'doi1' },
      { material: 'data', value: 'doi2' },
    ]);
    const { asFragment } = render(<DOIListShowAll dois={dois} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no dataDois', () => {
    const { asFragment } = render(<DOIListShowAll dois={fromJS([])} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
