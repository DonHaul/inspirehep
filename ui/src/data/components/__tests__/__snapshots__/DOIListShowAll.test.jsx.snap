// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOIListShowAll renders correctly with data and other dois 1`] = `
<DocumentFragment>
  <div
    class="ant-row"
  >
    <div
      class="ant-col"
    >
      <div
        class="__InlineList__"
        data-testid="inline-data-list"
      >
        <span
          class=""
        >
          DOI: 
        </span>
        <ul>
          <li>
            <span>
              <a
                href="//doi.org/doi1"
                target="_blank"
              >
                doi1
              </a>
              <span>
                 (data)
              </span>
            </span>
            , 
          </li>
          <li>
            <span>
              <a
                href="//doi.org/doi2"
                target="_blank"
              >
                doi2
              </a>
              <span>
                 (data)
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="di pl1"
    >
      <button
        class="__SecondaryButton__"
        type="button"
      >
        Show All(3)
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DOIListShowAll renders correctly with dataDois only 1`] = `
<DocumentFragment>
  <div
    class="ant-row"
  >
    <div
      class="ant-col"
    >
      <div
        class="__InlineList__"
        data-testid="inline-data-list"
      >
        <span
          class=""
        >
          DOI: 
        </span>
        <ul>
          <li>
            <span>
              <a
                href="//doi.org/doi1"
                target="_blank"
              >
                doi1
              </a>
              <span>
                 (data)
              </span>
            </span>
            , 
          </li>
          <li>
            <span>
              <a
                href="//doi.org/doi2"
                target="_blank"
              >
                doi2
              </a>
              <span>
                 (data)
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DOIListShowAll renders correctly with default props 1`] = `<DocumentFragment />`;

exports[`DOIListShowAll renders correctly with no dataDois 1`] = `<DocumentFragment />`;
